### PR TITLE
Added authentication and private routes

### PR DESCRIPTION
**Pull request by nikhiljugale007**
## PREVIEW: https://deploy-preview-8--genuine-cocada-362283.netlify.app
added authentication with private routes.
files to check:
app.js
RequireAuth.js
Login.jsx
Singup.jsx
Profile.jsx
VideoCard.js

### CODE REVIEW

File: src/pages/authentication/Signup.jsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import { Link } from "react-router-dom";
+import { CheckboxInput, FormInput } from "../../components";
+import { useState } from "react";
+import { validateForm } from "./FormValidator";
+// import { signupuser } from "../../../api/apicall";
```
Comment (yaswanthmyneni) [inline src/pages/authentication/Signup.jsx]: changes: remove unused code

File: src/pages/authentication/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	);
+	const [formError, setFormError] = useState(initialSignUpFormState);
+	let navigate = useNavigate();
+
+	const inputChangeHandler = (e) => {
+		const new_val = { [e.target.name]: e.target.value };
+		setsignUpFormState((prev) => ({ ...prev, ...new_val }));
+	};
+	const checkboxChangeHandler = (e) => {
+		setsignUpFormState((prev) => ({
```
Comment (yaswanthmyneni) [inline src/pages/authentication/Signup.jsx]: suggestion: setsignUpFormState should be like ------> setSignUpFormState

**Review by yaswanthmyneni — APPROVED** [review]
The code is clean and the separate validation function is good, you can merge the pr.

**Review by PuneetP16 — APPROVED** [review]
UI is good. good to merge.

UI suggestion: You could give some space between header and main section.

image

File: src/pages/authentication/authentication.css [hunk 3]
```diff
@@ -0,0 +1,23 @@
+.form-container {
+	display: flex;
+	flex-direction: column;
+	align-items: center;
+	justify-content: center;
+}
+.form {
+	max-width: 40rem;
+	width: 100%;
```
Comment (PuneetP16) [inline src/pages/authentication/authentication.css:9]: Suggestion: give spacing between two CSS classes for readability.

**Review by PuneetP16 — APPROVED** [review]

**Review by Sahil25061999 — APPROVED** [review]
everything looks good, auth functionality are working properly, nothing much to suggest.

File: src/pages/authentication/Signup.jsx (hunk 1 above)
Comment (nikhiljugale007) [reply src/pages/authentication/Signup.jsx]: removed.

File: src/pages/authentication/Signup.jsx (hunk 2 above)
Comment (nikhiljugale007) [reply src/pages/authentication/Signup.jsx]: changed function name to setSignUpFormState

File: src/pages/authentication/authentication.css (hunk 3 above)
Comment (nikhiljugale007) [reply src/pages/authentication/authentication.css:9]: will try to format my code more properly.

**Review by keyurshah339 — COMMENTED** [review]
clean code good to go

**Pull request merged**